Source/ASNetworkImageNode.h

```diff
@@ -196,6 +201,28 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)imageNodeDidFinishDecoding:(ASNetworkImageNode *)imageNode;
 
+/**
+ * Notification that the image node finished downloading an image, with additional info.
+ * If implemented, this method will be called instead of `imageNode:didFetchImage:`.
+ *
+ * @param imageNode The sender.
+ * @param image The newly-loaded image.
+ * @param info Additional information about the image load.
+ *
+ * @discussion Called on a background queue.
+ */
+- (void)imageNode:(ASNetworkImageNode *)imageNode didFetchImage:(UIImage *)image info:(ASNetworkImageLoadInfo *)info;
+
+/**
+ * Notification that the image node failed to download the image.
+ *
+ * @param imageNode The sender.
+ * @param error The error with details.
+ *
+ * @discussion Called on a background queue.
+ */
+- (void)imageNode:(ASNetworkImageNode *)imageNode didFailFetchingWithError:(NSError *)error;
```

Comment:
Now that we have this flag, we can continue using the old methods and remove these.

We also need to update the documentation of the existing methods to say:

`@discussion Called on the main thread if delegateCallbacksOnMainThread=YES (the default,) otherwise on a background thread.`

Reply:
I thought the goal is to remove the delegate to main ones eventually. If we use same call out block. Would it be harder handle the deprecate marks?